### PR TITLE
Fixes #2

### LINKED ISSUE #2: Right click menu doesn't work under some conditions
Right clicking causes the window to lock up with the current usage of Swing and AWT.

### PR DESCRIPTION
**Pull request by hecksadecimal**
Changes currPopup to JPopupMenu and implements with the canvas click check to hide the menu when you click elsewhere. Additionally, currPopup is set to Heavyweight mode and added to the Canvas' parent to sidestep some heavyweight/lightweight incompatibility issues.

### CODE REVIEW

File: src/monster860/fastdmm/FastDMM.java [hunk 1]
```diff
@@ -482,27 +482,33 @@ else if(dwheel < 0)
 		
 		while(Mouse.next()) {
 			if(Mouse.getEventButtonState()) {
+				try {
+					currPopup.setVisible(false);
+				} catch (NullPointerException e) {
+					// Doesn't exist. Oh well.
+				}
```
Comment (monster860) [inline src/monster860/fastdmm/FastDMM.java:491]: Please do this properly. like this:

```
if(currPopup != null {
    currPopup.setVisible(false);
    continue;
}
```

File: src/monster860/fastdmm/FastDMM.java [hunk 2]
```diff
@@ -482,27 +482,33 @@ else if(dwheel < 0)
...
 							item.addActionListener(new DeleteListener(this, l, i));
 							menu.add(item);
 							
-							item = new MenuItem("View Variables");
+							item = new JMenuItem("View Variables");
 							item.addActionListener(new EditVarsListener(this, l, i));
 							menu.add(item);
 						}
-						canvas.add(currPopup);
+						canvas.getParent().add(currPopup);
```
Comment (monster860) [inline src/monster860/fastdmm/FastDMM.java:513]: Have you tested this to make sure the right click menu actually shows up in the correct place?

**Comment by monster860** [comment]
Please do something about the 

`Merge remote-tracking branch 'refs/remotes/monster860/master'`

commits

File: src/monster860/fastdmm/FastDMM.java (hunk 2 above)
Comment (hecksadecimal) [inline src/monster860/fastdmm/FastDMM.java:513]: I have, yes. It behaves as expected.

**Comment by hecksadecimal** [comment]
How do you propose I do that? It happens whenever I pull from this repo to get the latest changes.

File: src/monster860/fastdmm/FastDMM.java (hunk 1 above)
Comment (hecksadecimal) [inline src/monster860/fastdmm/FastDMM.java:491]: That changes the behavior. Doing this causes it to disappear correctly the first time, but then it no longer creates the context menu on right click.

**Comment by monster860** [comment]
How to fix having these fucked up merge commits on your `master` branch:

```
git fetch monster860
git checkout master
git reset --hard HEAD~<number of merge commits>  (If you have 4 merge commits do git reset --hard HEAD~4
git merge monster860/master --ff-only
git push --force origin master
```

How to fix having these fucked up merge commis on your `Fix-Context` branch:
(Do this after fixing it on your master branch)

```
git checkout
git reset --soft HEAD~
git stash
git reset --hard HEAD~2
git merge master
git stash apply
git push --force origin Fix-Context
```

Do not use the `Update from monster860/master` button in Github. It sucks balls.

Instead, do this in the command line whenever you want to update

```
git fetch monster860
git checkout master
git merge monster860/master --ff-only
```

File: src/monster860/fastdmm/FastDMM.java (hunk 1 above)
Comment (monster860) [inline src/monster860/fastdmm/FastDMM.java:491]: Put this before the if statement

```
if(currPopup != null && !currPopup.isVisible())
    currPopup = null;
```

And inside, between the setvisible and continue call put `currPopup = null`

**Comment by monster860** [comment]
Oh yeah, I forgot to mention you need to actually commit your changes and then you can reopen _this_ pull request.

**Comment by hecksadecimal** [comment]
Alrighty. I'm trying your code change suggestions right now.

**Comment by hecksadecimal** [comment]
Applied and tested the changes. Behaves as expected.

**Pull request merged**